xbmc/windowing/X11/GLContextEGL.cpp

```diff
@@ -98,12 +98,14 @@ bool CGLContextEGL::Refresh(bool force, int screen, Window glWindow, bool &newCo
     return false;
   }
 
+#if defined(HAS_GL)
```

Comment:
Please avoid ifdefs if possible. It would be better to have the API as function argument or similar. Actually, using `CEGLContextUtils` would also solve this and remove a lot of the code duplicated in this file. It would be a bit more effort though and also touch the GL side of things (which we would welcome). We can give you more information if you want to go down that route.

Reply:
The ifdef seemed to me the best approach here, because it had the least effect on existing code. 
Btw, please note that almost all of the changes in this PR are simply copied from the Krypton branch, with minimal adaptions. 
(As I wrote before, I'll be away now until the beginning of November, so please be patient)

Reply:
see commit baa4e7943351855bbad05a94d4d037d3c264fe3a